### ISSUE TITLE
Unable to process: Found an unexpected BAM tag data type: [B] while looking for a tag (XD)

### ISSUE THREAD

**Issue by ShannonDaddy** [Illumina/Pisces#13]
I found an exception of Pisces when one of my sam record is like this：
cons:1305       113     chr1    115256480       60      112M1D23M       =       115256480       -1      CCNTNCGNCTG NN#N#NN#NNNNN#NNNNNNNNNN MC:Z:112M1D23M  MD:Z:0G1C1T2C5C27A20C49^T21G1   cD:i:7  NM:i:9  cM:i:7  MQ:i:60 UQ:i:59 AS:i:113        RX:Z:ACACCAATCAATCTAGACACCAATCAATCTAG   cd:B:s,7,7,7,7,7,7

But in the pysam document, the data type B is supported:
A: printable char
i: signed int
f: float
Z: printable string
H: Byte array in hex format
B: Integer or numeric array

I wonder what kind of data types Pisces supports? 
Thanks a lot!

**Comment by tamsen** [comment]
Yeah, that's not expected. Whats your command line? Was this BAM produced by the stitcher, and was the same Pisces version used for the Stitcher and for Pisces? What version was it?

**Comment by ShannonDaddy** [comment]
No, I didn't use stitcher to produce the BAM. It was produced by fgbio FilterConsensusReads. I got a way to solve this problem by remove all the information of type B type. Here is part of my code:
` $fgbio FilterConsensusReads -i $prefix.tag.namesorted.mq.grouped.cons.bam \
   -o $prefix.tag.namesorted.mq.grouped.cons.flt.bam \
   --ref=$ref \
   --reverse-per-base-tags=true \
   --min-reads=2 \
   -E 0.05 \
   -N 40 \
   -e 0.1 \
   -n 0.1

   #5. align consensus sequences
   $picard SortSam I=$prefix.tag.namesorted.mq.grouped.cons.flt.bam O=$prefix.cons.unmapped.bam SO=queryname
   $picard SamToFastq I=$prefix.cons.unmapped.bam \
   F=/dev/stdout INTERLEAVE=true \
   | $bwa mem -p -t 2 $ref /dev/stdin \
   | $picard MergeBamAlignment \
   UNMAPPED=$prefix.cons.unmapped.bam ALIGNED=/dev/stdin \
   o=$prefix.cons.mapped.bam R=$ref \
   SO=coordinate ALIGNER_PROPER_PAIR_FLAGS=true MAX_GAPS=-1 \
   VALIDATION_STRINGENCY=SILENT CREATE_INDEX=true

   #6. Variant Calling by Pisces
   $samtools view -h $prefix.cons.mapped.bam|\
   sed 's/cd:B:/cd:i:/' |\
   sed 's/ce:B:/ce:i:/'|\
   $samtools view -bS - >$prefix.cons.mapped.adj.bam
   $samtools index $prefix.cons.mapped.adj.bam
   sh scripts/Pisces.sh $prefix.cons.mapped.adj.bam
`

**Comment by tamsen** [comment]
I'm glad you found a work around.  But I'm not happy about the behavior. We'll get back to you on this.

**Comment by GWheelerEB** [comment]
Hello. Has there been an update on this? I am currently trying to use Pisces ver. 5.2.9.122 and am getting the '[B] while looking for a tag (XD)' error on every .bam file I have tested. The BAMs are created as part of a pipeline including BWA 0.7,15, GATK 4.0.5, and Samblaster 0.1.22.

I am able to run Pisces on these .bams if I skip the Stitcher step entirely, but I'm not sure if that deviates greatly from the program's expectations and whether or not it would increase the odds of erroneous results.
